### PR TITLE
[src] Move enums into files that are processed by the generator

### PR DESCRIPTION
**Pull request by spouliot**
The generator has been able to read and re-generate enums for a while.
However not all enums were in file processed by the generator. That was
not a huge problem since the output was identical, unless this was a
"smart enum".

Supporting .net5+ `[Uns|S]upportedOSPlatformAttribute` on API requires
a lot of changes. Generated bindings are much easier since the generator
will output the new attributes for dotnet and keep the older ones for
_legacy_.

So moving the enums to generated code means a lot less code to update
for dotnet - and means additional enums (new Xcode) won't have to include
both new and legacy attributes.

The number of manual updates (for iOS) is about 1500 less (5792 down to
4219) than before.

Helps https://github.com/spouliot/xamarin-macios/pull/new/gh10170

### CODE REVIEW

**Review by dalexsoto — APPROVED** [review]
yay! ♥️

File: src/AudioUnit/AUEnums.cs [hunk 1]
```diff
@@ -0,0 +1,910 @@
...
+		Distance = 2, 
+		Gain = 3, 
+		PlaybackRate = 4,
+		Enable = 5,
+		MinGain = 6, 
+		MaxGain = 7,
+		ReverbBlend = 8,
+		GlobalReverbGain = 9,
+		OcclussionAttenuation = 10,
+		ObstructionAttenuation = 11
```
Comment (rolfbjarne) [inline src/AudioUnit/AUEnums.cs:632]:
```suggestion
		ObstructionAttenuation = 11,
```

File: src/AudioUnit/AUEnums.cs [hunk 2]
```diff
@@ -0,0 +1,910 @@
...
+	}
+
+	[iOS (8, 0)]
+	public enum SpatialMixerAttenuation {
+		Power = 0,
+		Exponential = 1,
+		Inverse = 2,
+		Linear = 3
```
Comment (rolfbjarne) [inline src/AudioUnit/AUEnums.cs:640]:
```suggestion
		Linear = 3,
```

File: src/AudioUnit/AUEnums.cs [hunk 3]
```diff
@@ -0,0 +1,910 @@
...
+
+		[iOS (8, 0)]
+		[Mac (10, 10)]
+		Loop                   = 0x08,
+		[iOS (8, 0)]
+		[Mac (10, 10)]
+		Interrupt              = 0x10,
+		[iOS (8, 0)]
+		[Mac (10, 10)]
+		InterruptAtLoop        = 0x20
```
Comment (rolfbjarne) [inline src/AudioUnit/AUEnums.cs:665]:
```suggestion
		InterruptAtLoop        = 0x20,
```

File: src/AudioUnit/AUEnums.cs [hunk 4]
```diff
@@ -0,0 +1,910 @@
...
+
+	public enum AudioUnitScopeType { // UInt32 AudioUnitScope
+		Global		= 0,
+		Input		= 1,
+		Output		= 2,
+		Group		= 3,
+		Part		= 4,
+		Note		= 5,
+		Layer		= 6,
+		LayerItem	= 7
```
Comment (rolfbjarne) [inline src/AudioUnit/AUEnums.cs:676]:
```suggestion
		LayerItem	= 7,
```

File: src/AudioUnit/AUEnums.cs [hunk 5]
```diff
@@ -0,0 +1,910 @@
...
+	[Flags]
+	public enum AudioUnitRenderActionFlags { // UInt32 AudioUnitRenderActionFlags
+		PreRender = (1 << 2),
+		PostRender = (1 << 3),
+		OutputIsSilence = (1 << 4),
+		OfflinePreflight = (1 << 5),
+		OfflineRender = (1 << 6),
+		OfflineComplete = (1 << 7),
+		PostRenderError = (1 << 8),
+		DoNotCheckRenderArgs = (1 << 9)
```
Comment (rolfbjarne) [inline src/AudioUnit/AUEnums.cs:688]:
```suggestion
		DoNotCheckRenderArgs = (1 << 9),
```

File: src/AudioUnit/AUEnums.cs [hunk 6]
```diff
@@ -0,0 +1,910 @@
...
+	}
+
+	public enum AudioUnitRemoteControlEvent // Unused?
+	{
+		TogglePlayPause		= 1,
+		ToggleRecord		= 2,
+		Rewind				= 3
```
Comment (rolfbjarne) [inline src/AudioUnit/AUEnums.cs:695]:
```suggestion
		Rewind				= 3,
```

File: src/AudioUnit/AUEnums.cs [hunk 7]
```diff
@@ -0,0 +1,910 @@
...
+	}
+
+	[Native]
+	public enum AudioUnitBusType : long
+	{
+		Input = 1,
+		Output = 2
```
Comment (rolfbjarne) [inline src/AudioUnit/AUEnums.cs:702]:
```suggestion
		Output = 2,
```

File: src/AudioUnit/AUEnums.cs [hunk 8]
```diff
@@ -0,0 +1,910 @@
...
+	}
+
+	[Native]
+	public enum AUHostTransportStateFlags : ulong
+	{
+		Changed = 1,
+		Moving = 2,
+		Recording = 4,
+		Cycling = 8
```
Comment (rolfbjarne) [inline src/AudioUnit/AUEnums.cs:711]:
```suggestion
		Cycling = 8,
```

File: src/AudioUnit/AUEnums.cs [hunk 9]
```diff
@@ -0,0 +1,910 @@
...
+	{
+		Immediate = unchecked ((long) 0xffffffff00000000)
+	}
+
+	public enum AURenderEventType : byte
+	{
+		Parameter = 1,
+		ParameterRamp = 2,
+		Midi = 8,
+		MidiSysEx = 9
```
Comment (rolfbjarne) [inline src/AudioUnit/AUEnums.cs:724]:
```suggestion
		MidiSysEx = 9,
```

File: src/AudioUnit/AUEnums.cs [hunk 10]
```diff
@@ -0,0 +1,910 @@
...
+	{
+		Changed = 1,
+		Moving = 2,
+		Recording = 4,
+		Cycling = 8
+	}
+
+	public enum AUEventSampleTime : long
+	{
+		Immediate = unchecked ((long) 0xffffffff00000000)
```
Comment (rolfbjarne) [inline src/AudioUnit/AUEnums.cs:716]:
```suggestion
		Immediate = unchecked ((long) 0xffffffff00000000),
```

File: src/AudioUnit/AUEnums.cs [hunk 11]
```diff
@@ -0,0 +1,910 @@
...
+		Parameter = 1,
+		ParameterRamp = 2,
+		Midi = 8,
+		MidiSysEx = 9
+	}
+
+	[iOS (9,0), Mac (10,11)]
+	public enum AudioComponentInstantiationOptions : uint {
+		OutOfProcess = 1,
+		InProcess = 2
```
Comment (rolfbjarne) [inline src/AudioUnit/AUEnums.cs:730]:
```suggestion
		InProcess = 2,
```

File: src/AudioUnit/AUEnums.cs [hunk 12]
```diff
@@ -0,0 +1,910 @@
...
+		DisplayLogarithmic = (1 << 22),
+		IsHighResolution = (1 << 23),
+		NonRealTime = (1 << 24),
+		CanRamp = (1 << 25),
+		ExpertMode = (1 << 26),
+		HasCFNameString = (1 << 27),
+		IsGlobalMeta = (1 << 28),
+		IsElementMeta = (1 << 29),
+		IsReadable = (1 << 30),
+		IsWritable = unchecked((uint)1 << 31)
```
Comment (rolfbjarne) [inline src/AudioUnit/AUEnums.cs:763]:
```suggestion
		IsWritable = unchecked((uint)1 << 31),
```

File: src/AudioUnit/AUEnums.cs [hunk 13]
```diff
@@ -0,0 +1,910 @@
...
+		[iOS (14,0)][TV (14,0)][Mac (11,0)]
+		UseOutputType = 7,
+	}
+
+	public enum AU3DMixerAttenuationCurve : uint
+	{
+		Power = 0,
+		Exponential = 1,
+		Inverse = 2,
+		Linear = 3
```
Comment (rolfbjarne) [inline src/AudioUnit/AUEnums.cs:840]:
```suggestion
		Linear = 3,
```

File: src/AudioUnit/AUEnums.cs [hunk 14]
```diff
@@ -0,0 +1,910 @@
...
+	}
+
+	public enum AUSpatialMixerRenderingFlags : uint
+	{
+		InterAuralDelay = (1 << 0),
+		DistanceAttenuation = (1 << 2)
```
Comment (rolfbjarne) [inline src/AudioUnit/AUEnums.cs:846]:
```suggestion
		DistanceAttenuation = (1 << 2),
```

File: src/AudioUnit/AUEnums.cs [hunk 15]
```diff
@@ -0,0 +1,910 @@
...
+	}
+
+	[iOS (10,0), Mac (10,12)]
+	public enum AUParameterAutomationEventType : uint {
+		Value = 0,
+		Touch = 1,
+		Release = 2
```
Comment (rolfbjarne) [inline src/AudioUnit/AUEnums.cs:853]:
```suggestion
		Release = 2,
```

File: src/AudioUnit/AUEnums.cs [hunk 16]
```diff
@@ -0,0 +1,910 @@
...
+	}
+
+	public enum AUScheduledAudioSliceFlags : uint
+	{
+		Complete = 1,
+		BeganToRender = 2,
+		BeganToRenderLate = 4,
+		Loop = 8,
+		Interrupt = 16,
+		InterruptAtLoop = 32
```
Comment (rolfbjarne) [inline src/AudioUnit/AUEnums.cs:819]:
```suggestion
		InterruptAtLoop = 32,
```

File: src/AudioUnit/AUEnums.cs [hunk 17]
```diff
@@ -0,0 +1,910 @@
...
+		MediumHall = 3,
+		LargeHall = 4,
+		Plate = 5,
+		MediumChamber = 6,
+		LargeChamber = 7,
+		Cathedral = 8,
+		LargeRoom2 = 9,
+		MediumHall2 = 10,
+		MediumHall3 = 11,
+		LargeHall2 = 12
```
Comment (rolfbjarne) [inline src/AudioUnit/AUEnums.cs:809]:
```suggestion
		LargeHall2 = 12,
```

File: src/AudioUnit/AUEnums.cs [hunk 18]
```diff
@@ -0,0 +1,910 @@
...
+
+	public enum AU3DMixerRenderingFlags : uint
+	{
+		InterAuralDelay = (1 << 0),
+		DopplerShift = (1 << 1),
+		DistanceAttenuation = (1 << 2),
+		DistanceFilter = (1 << 3),
+		DistanceDiffusion = (1 << 4),
+		LinearDistanceAttenuation = (1 << 5),
+		ConstantReverbBlend = (1 << 6)
```
Comment (rolfbjarne) [inline src/AudioUnit/AUEnums.cs:792]:
```suggestion
		ConstantReverbBlend = (1 << 6),
```

File: src/AudioUnit/AUEnums.cs [hunk 19]
```diff
@@ -0,0 +1,910 @@
...
+		UnauthorizedErrorOpen,
+		UnauthorizedErrorInit
+	}
+
+	public enum AUSpatialMixerAttenuationCurve : uint
+	{
+		Power = 0,
+		Exponential = 1,
+		Inverse = 2,
+		Linear = 3
```
Comment (rolfbjarne) [inline src/AudioUnit/AUEnums.cs:781]:
```suggestion
		Linear = 3,
```

File: src/AudioUnit/AUEnums.cs [hunk 20]
```diff
@@ -0,0 +1,910 @@
...
+	}
+
+	public enum AudioComponentValidationResult : uint
+	{
+		Unknown = 0,
+		Passed,
+		Failed,
+		TimedOut,
+		UnauthorizedErrorOpen,
+		UnauthorizedErrorInit
```
Comment (rolfbjarne) [inline src/AudioUnit/AUEnums.cs:773]:
```suggestion
		UnauthorizedErrorInit,
```

File: src/AudioUnit/AUEnums.cs [hunk 21]
```diff
@@ -0,0 +1,910 @@
...
+	public enum AudioComponentInstantiationOptions : uint {
+		OutOfProcess = 1,
+		InProcess = 2
+	}
+
+	[Native]
+	public enum AUAudioUnitBusType : long
+	{
+		Input = 1,
+		Output = 2
```
Comment (rolfbjarne) [inline src/AudioUnit/AUEnums.cs:737]:
```suggestion
		Output = 2,
```

**Review by rolfbjarne — CHANGES_REQUESTED** [review]
The important changes here is that structs shouldn't have a different size between COREBUILD and not, because then the generator might generate the wrong code. All the other changes are just cosmetic and not really important.

File: src/CoreGraphics/CGAffineTransform.cs [hunk 22]
```diff
@@ -37,6 +37,7 @@ namespace CoreGraphics {
 	// CGAffineTransform.h
 	[StructLayout(LayoutKind.Sequential)]
 	public struct CGAffineTransform {
+#if !COREBUILD
```
[the remark this answers is not in the document]
Comment (spouliot) [reply src/CoreGraphics/CGAffineTransform.cs:40]: I had to adjust a few `#if` for compilation issues but I don't recall that one in particular. Will fix...

**Review by rolfbjarne — APPROVED** [review]

**Review by tj-devel709 — APPROVED** [review]

**Comment by spouliot** [comment]
Failure is unrelated to PR

introspection/watchOS 32-bits - simulator/Debug (watchOS 3.2): Failed
* https://github.com/xamarin/maccore/issues/2378

**Pull request merged**